app/routers/temperatures.py

```
@@ -0,0 +1,91 @@
+from datetime import datetime, timezone
+from typing import List, Optional
+
+import httpx
+from fastapi import APIRouter, Depends, HTTPException, Query, status
+from sqlalchemy.orm import Session
+
+from .. import models, schemas
+from ..deps import get_db
+
+router = APIRouter(prefix="/temperatures", tags=["temperatures"])
+
+GEOCODING_URL = "https://geocoding-api.open-meteo.com/v1/search"
+WEATHER_URL = "https://api.open-meteo.com/v1/forecast"  # using
+
+
+# current_weather=true
+
+
+async def _fetch_city_temperature(
+        client: httpx.AsyncClient,
+        city: models.City
+) -> Optional[float]:
+    """Resolve city name to lat/lon, then fetch current temperature in В°C.
+    Returns None if the city can't be resolved or the API returns no value.
+    """
+    try:
+        # 1) geocode
+        resp = await client.get(
+            GEOCODING_URL,
+            params={"name": city.name, "count": 1}
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        if not data or not data.get("results") or len(data["results"]) == 0:
+            return None
+        lat = data["results"][0]["latitude"]
+        lon = data["results"][0]["longitude"]
+        # 2) weather (classic param)
+        w = await client.get(WEATHER_URL, params={
+            "latitude": lat,
+            "longitude": lon,
+            "current_weather": True,
+            "timezone": "UTC",
+        })
+        w.raise_for_status()
+        wd = w.json()
+        # Parse classic current_weather.temperature (in В°C)
+        cw = wd.get("current_weather") or {}
+        temp = cw.get("temperature")
+        return float(temp) if temp is not None else None
+    except Exception:
+        return None
```

The except block here swallows all exceptions and simply returns None. This hides failures from logs and makes debugging harder. Catch more specific exceptions (for example `httpx.HTTPError`) and log details (city name, error message). You can still return None for failures, but log the reason so operators can diagnose issues.